src/tools/perf/lib/libperf.c

```diff
@@ -1257,6 +1347,29 @@ static ucs_status_t ucp_perf_test_flush_workers(ucx_perf_context_t *perf)
     return ctx.status;
 }
 
+static ucs_status_t ucp_perf_test_setup_self_endpoints(ucx_perf_context_t *perf)
+{
+    int i;
+    ucs_status_t status;
+
+    for (i = 0; i < perf->params.thread_count; ++i) {
+        status = ucp_perf_test_setup_self_endpoint(&perf->ucp.tctx[i].perf, i);
```

Comment:
we can unite ucp_perf_test_setup_self_endpoint into ucp_perf_test_setup_self_endpoints - since need to get worker address only once, and ep_params are the same, it can reduce overall amount of code

Reply:
BTW, maybe we can create self endpoints only if we have non-host memory?

Reply:
Can I use worker and address of tctx[0] for all of the eps?